server/sonar-server/src/test/java/org/sonar/server/usergroups/ws/UpdateActionTest.java

```diff
@@ -60,24 +61,27 @@
 
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
-  DbClient dbClient = db.getDbClient();
-  DbSession dbSession = db.getSession();
-  GroupDao groupDao = dbClient.groupDao();
-  UserGroupDao userGroupDao = dbClient.userGroupDao();
 
-  PersistentSettings settings = mock(PersistentSettings.class);
-  WsTester ws = new WsTester(new UserGroupsWs(new UpdateAction(dbClient, userSession, new UserGroupUpdater(dbClient), settings)));
+  @Rule
+  public DefaultOrganizationProviderRule defaultOrganizationProvider = DefaultOrganizationProviderRule.create(db);
+
+  private DbClient dbClient = db.getDbClient();
+  private DbSession dbSession = db.getSession();
+  private PersistentSettings settings = mock(PersistentSettings.class);
+  private WsTester ws;
 
   @Before
   public void setUp() throws Exception {
+    GroupWsSupport groupSupport = new GroupWsSupport(db.getDbClient(), defaultOrganizationProvider);
+    ws = new WsTester(new UserGroupsWs(new UpdateAction(dbClient, userSession, groupSupport, settings)));
```

Comment:
This is a subjective topic. The reason is that I'm not a big fan of long assignment in a declaration, even if it's on multiple lines. I prefer splitting assignment, so for example in `@Before` method.

In this example your proposal is to have:
```
private WsTester ws = new WsTester(new UserGroupsWs(new UpdateAction(db.getDbClient(), userSession, new GroupWsSupport(db.getDbClient(), defaultOrganizationProvider), settings, defaultOrganizationProvider)));
```